src/Core/Framework/DataAbstractionLayer/Search/Aggregation/Metric/RangeAggregation.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopware\Core\Framework\DataAbstractionLayer\Search\Aggregation\Metric;
+
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Aggregation\Aggregation;
+
+/**
+ * Allow to aggregate data on predefined ranges values
+ *
+ * Bound are computed as in the elasticsearch range aggregation :
+ *    the "from" will be compared with greater than or equal
+ *    the "to" will be compared with lower than
+ *
+ * @final tag:v6.5.0
+ */
+class RangeAggregation extends Aggregation
+{
+    /**
+     * @var array<int, array<string, float|string|null>>
+     */
+    protected array $ranges;
+
+    /**
+     * @param array<int, array<string, float|string|null>> $ranges
+     */
+    public function __construct(string $name, string $field, array $ranges)
+    {
+        parent::__construct($name, $field);
+        foreach ($ranges as $range) {
+            $this->addRange(
+                isset($range['from']) ? (float) $range['from'] : null,
+                isset($range['to']) ? (float) $range['to'] : null,
+                isset($range['key']) ? (string) $range['key'] : null,
+            );
+        }
+    }
+
+    /**
+     * @return array<int, array<string, float|string|null>>
+     */
+    public function getRanges(): array
+    {
+        return $this->ranges;
+    }
+
+    /**
+     * Add a new range in the aggregation.
+     * If no key is provided, the key will be build using the $from and $to bounds ($this->buildRangeKey())
+     *
+     * @param ?float $from
+     * @param ?float $to
+     * @param ?string $key
+     *
+     * @return $this
```

Comment:
You can remove phpdoc which doesn't add more value than the actual method parameter types